base/set.jl

```diff
@@ -17,6 +17,46 @@ Set() = Set{Any}()
 Construct a [`Set`](@ref) of the values generated by the given iterable object, or an
 empty set. Should be used instead of [`BitSet`](@ref) for sparse integer sets, or
 for sets of arbitrary objects.
+
+# Examples
+```jldoctest
+julia> animals = Set{String}(["cat","fish","dog","bird"])
+Set{String} with 4 elements:
+  "fish"
+  "cat"
+  "bird"
+  "dog"
+```
+
+If the element type is not given, it is computed by intersecting the types of the arguments:
```

Comment:
Would be good to have an example where not all elements are of the same type here, like

```
julia> Set([2, 1.0, Int8(3)])
Set{Float64} with 3 elements:
  2.0
  3.0
  1.0
```